pennylane/qnn/cost.py

```diff
@@ -0,0 +1,115 @@
+"""
+This submodule contains frequently used loss and cost functions.
+"""
+# pylint: disable=too-many-arguments, too-few-public-methods
+import pennylane as qml
+
+
+class SquaredErrorLoss:
+    r"""Combines an ansatz circuit with some target observables and calculates
+    the squared error between their expectation values and a target.
+
+    Args:
+        ansatz (callable): The ansatz for the circuit before the final measurement step.
+            Note that the ansatz **must** have the following signature:
+
+            .. code-block:: python
+
+                ansatz(params, **kwargs)
+
+            where ``params`` are the trainable weights of the variational circuit, and
+            ``kwargs`` are any additional keyword arguments that need to be passed
+            to the template.
+        observables (Iterable[:class:`~.Observable`]): observables to measure during the
+            final step of each circuit
+        device (Device, Sequence[Device]): Corresponding device(s) where the resulting
+            function should be executed. This can either be a single device, or a list
+            of devices of length matching the number of observables.
+        interface (str, None): Which interface to use.
+            This affects the types of objects that can be passed to/returned to the function.
+            Supports all interfaces supported by the :func:`~.qnode` decorator.
+        diff_method (str, None): The method of differentiation to use with the created function.
+            Supports all differentiation methods supported by the :func:`~.qnode` decorator.
+
+    Returns:
+        callable: a loss function with signature ``loss(*args, target=None, **kwargs)`` that calculates
+        the squared error loss between the observables' expectation values and a target.
+
+    .. seealso:: :func:`~.map`
+
+    **Example:**
+
+    First, we create a device and design an ansatz:
+
+    .. code-block:: python
+
+        dev = qml.device('default.qubit', wires=3)
+
+        def ansatz(phis, **kwargs):
+            for w, phi in enumerate(phis):
+                qml.RX(phi, wires=w)
```

Comment:
Very minor, but how about a CNOT connecting from 3 to 1 and 2? Just so that the third qubit does something, since our observables are only on the first and second qubit.

Reply:
Sure, I could add it to make it more "interesting", but isn't the third qubit used in the last observable? Or do I misunderstand your comment?

Reply:
:facepalm: Nope, you're right! It's good as is :grin: